gourmet/exporters/exporter.py

```diff
@@ -296,15 +291,23 @@ def handle_markup (self, txt):
             return ''
         from gi.repository import Pango
         outtxt = ""
+        # The Python interface to pango wants a str, but also a length, which
+        # I believe is the number of bytes in the UTF-8 representation.
+        b = txt.encode('utf-8')
         try:
-            al,txt,sep = Pango.parse_markup(txt,'\x00')
-        except:
-            al,txt,sep = Pango.parse_markup(xml.sax.saxutils.escape(txt),'\x00')
+            ok, al, txt, sep = Pango.parse_markup(txt, len(b), '\x00')
+        except:  # FIXME: Catch a more specific error?
```

Comment:
This is a `g-markup-error-quark` due to unescaped text. The exception's type is `gi.repository.GLib.Error`

Reply:
here's an example:
```python
try: 
    Pango.parse_markup("&amp", -1, '\x00') 
except gi.repository.GLib.Error: 
    print('missing a semicolon for proper parsing')
```
It's also legal to use `-1` to denote null terminated strings.

Reply:
I couldn't quite work out if it was safe to rely on a null terminator of the bytes it got from Python. Python strings can *contain* null characters (`'\x00'`), but I can't imagine it ever making sense to have one in a string to be displayed. :-S

Reply:
Never mind, I experimented and it stops at a null byte even if you specify the size.

I think I've also satisfied myself that the obvious ways it could convert a Python `str` to a C `char*` pointer will give it a terminating NULL, so we can rely on that.